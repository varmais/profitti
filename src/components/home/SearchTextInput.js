import React, { Component } from 'react';
import {
  StyleSheet,
  TextInput
} from 'react-native';
import config from '../../config';

const styles = StyleSheet.create({
  input: {
    height: 40,
    paddingVertical: 4,
    paddingHorizontal: 6,
    marginTop: 16,
    marginBottom: 16,
    borderWidth: 1,
    borderColor: config.colors.green,
    borderRadius: 8,
    color: config.colors.white,
    fontSize: 18
  }
});

export default class SearchTextInput extends Component {
  render() {
    return (
      <TextInput
        style={styles.input}
        placeholder='Hae lauluja..'
        placeholderTextColor={config.colors.white}
        autoCorrect={false}
        returnKeyType='search'
        onSubmitEditing={this.handleSubmit}
      />
    );
  }

  handleSubmit = ({nativeEvent}) => {
    console.log(nativeEvent);
  };
}
