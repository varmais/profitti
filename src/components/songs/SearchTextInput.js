import React, { Component, PropTypes } from 'react';
import {
  StyleSheet,
  TextInput
} from 'react-native';
import { connect } from 'react-redux';
import config from '../../config';
import { searchSongs } from '../../redux/songs';

const styles = StyleSheet.create({
  input: {
    height: 40,
    paddingVertical: 4,
    paddingHorizontal: 8,
    marginVertical: 16,
    borderWidth: 1,
    borderColor: config.colors.grayMid,
    color: config.colors.blackLight,
    fontSize: 18
  }
});

export class SearchTextInput extends Component {
  static propTypes = {
    searchSongs: PropTypes.func.isRequired,
    navigation: PropTypes.shape({
      navigate: PropTypes.func.isRequired
    }).isRequired
  };

  render() {
    return (
      <TextInput
        style={styles.input}
        placeholder='Hae lauluja..'
        placeholderTextColor={config.colors.grayDark}
        autoCorrect={false}
        returnKeyType='search'
        onSubmitEditing={this.handleSubmit}
      />
    );
  }

  handleSubmit = ({nativeEvent}) => {
    this.props.searchSongs(nativeEvent.text);
    this.props.navigation.navigate('SongSearch', {searchString: nativeEvent.text});
  };
}

export default connect(null, {
  searchSongs
})(SearchTextInput);
