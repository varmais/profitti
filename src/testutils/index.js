import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';

global.React = React;
global.renderer = renderer;

export function createNavigationProp (params = {}) {
  return {
    state: {params: params},
    navigate: () => {}
  }
}