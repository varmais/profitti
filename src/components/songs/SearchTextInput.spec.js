import { createNavigationProp } from '../../testutils';
import { SearchTextInput } from './SearchTextInput';

describe('components:songs:SearchTextInput', () => {
  let searchSongsStub;

  beforeEach(() => {
    searchSongsStub = jest.fn();
  });

  test('renders SongButton', () => {
    const snapshot = renderer.create(<SearchTextInput
      searchSongs={searchSongsStub}
      navigation={createNavigationProp()}
    />).toJSON();
    expect(snapshot).toMatchSnapshot();
  });

  describe('when search is submitted', () => {
    test('it searches songs');
    test('it navigates to song list screen');
  });
});