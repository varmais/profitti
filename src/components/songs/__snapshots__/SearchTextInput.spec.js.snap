// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components:songs:SearchTextInput renders SongButton 1`] = `
<TextInput
  autoCorrect={false}
  onSubmitEditing={[Function]}
  placeholder="Hae lauluja.."
  placeholderTextColor="#414e51"
  returnKeyType="search"
  style={
    Object {
      "borderColor": "#caccce",
      "borderWidth": 1,
      "color": "#222",
      "fontSize": 18,
      "height": 40,
      "marginVertical": 16,
      "paddingHorizontal": 8,
      "paddingVertical": 4,
    }
  }
/>
`;
